src/IdentityServer8/src/ResponseHandling/Default/DiscoveryResponseGenerator.cs

```diff
@@ -331,7 +331,7 @@ where scope.ShowInDiscoveryDocument
         }
 
         // custom entries
-        if (!Options.Discovery.CustomEntries.IsNullOrEmpty())
+        if (!Options.Discovery.CustomEntries.Any())
```

Comment:
Should use the extension method EnumerableIsNullOrEmpty. In IdentityServer4 it called IsNullOrEmpty, but this causes a conflict in .NET 8, so the extension method was renamed EnumerableIsNullOrEmpty in IdentityServer8.